### ISSUE TITLE
Support Spring 5 and Spring Boot 2

### ISSUE THREAD

**Issue by omau** [zalando/riptide#284]
Since Spring Boot 1.5 will not support Java 9, we would like to change to Spring Boot 2 (M5 for now) as soon as possible. Unfortunately, Riptide does not seem to support it yet:

```
Caused by: java.lang.NoClassDefFoundError: org/springframework/boot/bind/PropertiesConfigurationFactory
    at org.zalando.riptide.spring.RiptidePostProcessor.setEnvironment (RiptidePostProcessor.java:24)
    at org.springframework.context.support.ApplicationContextAwareProcessor.invokeAwareInterfaces (ApplicationContextAwareProcessor.java:105)
    at org.springframework.context.support.ApplicationContextAwareProcessor.postProcessBeforeInitialization (ApplicationContextAwareProcessor.java:96)
    at org.springframework.beans.factory.supCaused by: java.lang.ClassNotFoundException: org.springframework.boot.bind.PropertiesConfigurationFactoryport.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization (AbstractAutowireCapableBeanFactory.java:423)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean (AbstractAutowireCapableBeanFactory.java:1696)
```

```
Caused by: java.lang.ClassNotFoundException: org.springframework.boot.bind.PropertiesConfigurationFactory
    at java.net.URLClassLoader.findClass (URLClassLoader.java:381)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:424)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:357)
    at org.zalando.riptide.spring.RiptidePostProcessor.setEnvironment (RiptidePostProcessor.java:24)
    at org.springframework.context.support.ApplicationContextAwareProcessor.invokeAwareInterfaces (ApplicationContextAwareProcessor.java:105)
```

**Comment by whiskeysierra** [comment]
Looks like Spring Boot 2.0 has some breaking changes. Maintaining compatibility with both versions will be hard. I'd rather go with a big switch, e.g. Riptide 3.0 that also bumps the Spring Boot version to 2.0. But I'll postpone this until Spring Boot 2.0 has been officially released.

**Comment by whiskeysierra** [comment]
@omau I just released 2.8.0 which adds support for Spring Boot 2.0
